src/Raven.Server/RavenServer.cs

```diff
@@ -350,19 +350,25 @@ private void RedirectsHttpTrafficToHttps()
                             if (response.IsSuccessStatusCode)
                             {
                                 // It worked, let's register this callback globally in the RequestExecutor
-                                if(RequestExecutor.HasServerCertificateCustomValidationCallback == false)
-                                    RequestExecutor.ServerCertificateCustomValidationCallback += CertificateCallback;
+                                if (RequestExecutor.HasServerCertificateCustomValidationCallback == false)
+                                {
+                                    var callbackTranslator = new RequestExecutor.CallbackTranslator
+                                    {
+                                        Callback = CertificateCallback
+                                    };
+                                    RequestExecutor.RemoteCertificateValidationCallback += callbackTranslator.Translate;
```

Comment:
Should not be translating here, change the signature of CertificateCallback

Reply:
yes